### ISSUE TITLE
Name resolution system

### ISSUE THREAD

**Issue by ripper234** [OmniLayer/spec#216]
[Motivation / context](https://github.com/mastercoin-MSC/omniwallet/issues/644).

The idea is to create a separate embedded system (optional to use) that maps a unique string name to a currency ID (or other entities in the system).

The use-case is just like domain names: It's nice to be able to type in something unique other than "Currency ID 1273015" to get to a currency. Instead think of "currency-cool-hoverboard-tokens-series-a".

The design for this should borrow from DNS, Namecoin, and Nxt's embedded "DNS-like" system.

**TL;DR**
1. Names are unique
2. Registering a name entry would burn X MSC.
3. Renewing a name entry would burn Y MSC
4. Mapping a name you control to an asset would burn Z MSC (we don't want people switching these pointers around all the time!)
5. If you don't renew an asset and it expires, everybody can register that.

**Things we need to decide:**
1. Is uniqueness case sensitive or insensitive? (advantage of insensitive - we don't really want two assets names that differ only by upper/lower case)
2. ...
